src/lib/resourceTiming.ts

```diff
@@ -0,0 +1,99 @@
+/*eslint guard-for-in:0*/
+import PerformanceObserver from "@fastly/performance-observer-polyfill";
+import compose from "../util/compose";
+import camelCaseToUnderscore from "../util/camelCaseToUnderscore";
+
+const EXCLUDED_PROPS = ["name", "initiatorType", "entryType"];
+
+function asyncGetEntry(
+  name: string,
+  timeout = 5000
+): Promise<ResourceTimingEntry> {
+  return new Promise(
+    (resolve, reject): void => {
+      let entry: PerformanceEntry | undefined;
+
+      const observer = new PerformanceObserver(
+        (
+          list: PerformanceObserverEntryList,
+          observer: PerformanceObserver
+        ): void => {
+          const namedEntries = list.getEntriesByName(name);
+          entry = namedEntries.shift();
+
+          if (entry) {
+            observer.disconnect();
+            resolve((entry as any) as ResourceTimingEntry);
+          }
+        }
+      );
+
+      setTimeout((): void => {
+        if (!entry) {
+          observer.disconnect();
+          reject(new Error("Timed out observing resource timing"));
+        }
+      }, timeout);
+
+      try {
+        observer.observe({ entryTypes: ["resource"] });
+      } catch (e) {
+        reject(e);
+      }
+    }
+  );
+}
+
+function cloneEntry(entry: ResourceTimingEntry): ResourceTimingEntry {
+  const result: ResourceTimingEntry = {};
+  for (const key in entry) {
+    const type = typeof entry[key];
+    if (type === "number" || type === "string") {
+      result[key] = entry[key];
+    }
+  }
+  return result;
+}
+
+function removeEntryProps(
```

Comment:
Can we use spread with this lib?
https://kangax.github.io/compat-table/es6/#test-spread_syntax_for_iterable_objects
If so - does this look better/cleaner?
```js
const {foo, bar, ...objWithRemovedProps} = originalObj;
```

Reply:
I'm going to create a ticket to spend some to focus on answering this question and generally cleaning up what is/isn't bundled via `preset-env`. So am going to skip in this PR.